beacon-chain/types/active_state.go

```diff
@@ -234,8 +256,32 @@ func (a *ActiveState) CalculateNewActiveState(
 		}
 	}
 
+	_, proposerIndex, err := casper.ProposerShardAndIndex(
+		cState.ShardAndCommitteesForSlots(),
+		cState.LastStateRecalculationSlot(),
+		parentSlot)
+	if err != nil {
+		return nil, fmt.Errorf("Can not get proposer index %v", err)
+	}
+
+	a.setRandaoMix(block.RandaoReveal())
```

Comment:
this should be a pure function

Reply:
I'm open to doing a deep copy of the active state at the top of this function as well, but a pure function is cleaner and (probably) more performant, so that's my preference. 

This is related to #663

Reply:
Ah! Good looking out!